libsigner/src/v0/messages.rs

```diff
@@ -0,0 +1,565 @@
+        assert_eq!(rejection, deserialized_rejection);
+
+        let rejection =
+            BlockRejection::new(Sha512Trunc256Sum([1u8; 32]), RejectCode::ConnectivityIssues);
+        let serialized_rejection = rejection.serialize_to_vec();
+        let deserialized_rejection = read_next::<BlockRejection, _>(&mut &serialized_rejection[..])
+            .expect("Failed to deserialize BlockRejection");
+        assert_eq!(rejection, deserialized_rejection);
+    }
+
+    #[test]
+    fn serde_block_response() {
+        let response =
+            BlockResponse::Accepted((Sha512Trunc256Sum([0u8; 32]), MessageSignature::empty()));
+        let serialized_response = response.serialize_to_vec();
+        let deserialized_response = read_next::<BlockResponse, _>(&mut &serialized_response[..])
+            .expect("Failed to deserialize BlockResponse");
+        assert_eq!(response, deserialized_response);
+
+        let response = BlockResponse::Rejected(BlockRejection::new(
+            Sha512Trunc256Sum([1u8; 32]),
+            RejectCode::ValidationFailed(ValidateRejectCode::InvalidBlock),
+        ));
+        let serialized_response = response.serialize_to_vec();
+        let deserialized_response = read_next::<BlockResponse, _>(&mut &serialized_response[..])
+            .expect("Failed to deserialize BlockResponse");
+        assert_eq!(response, deserialized_response);
+    }
+
+    #[test]
+    fn serde_signer_message() {
+        let signer_message = SignerMessage::BlockResponse(BlockResponse::Accepted((
+            Sha512Trunc256Sum([2u8; 32]),
+            MessageSignature::empty(),
+        )));
+        let serialized_signer_message = signer_message.serialize_to_vec();
+        let deserialized_signer_message =
+            read_next::<SignerMessage, _>(&mut &serialized_signer_message[..])
+                .expect("Failed to deserialize SignerMessage");
+        assert_eq!(signer_message, deserialized_signer_message);
+
+        let header = NakamotoBlockHeader {
+            version: 1,
+            chain_length: 2,
+            burn_spent: 3,
+            consensus_hash: ConsensusHash([0x04; 20]),
+            parent_block_id: StacksBlockId([0x05; 32]),
+            tx_merkle_root: Sha512Trunc256Sum([0x06; 32]),
+            state_index_root: TrieHash([0x07; 32]),
+            miner_signature: MessageSignature::empty(),
+            signer_signature: ThresholdSignature::empty(),
+            signer_bitvec: BitVec::zeros(1).unwrap(),
+        };
+        let mut block = NakamotoBlock {
+            header,
+            txs: vec![],
+        };
+        let tx_merkle_root = {
+            let txid_vecs = block
+                .txs
+                .iter()
+                .map(|tx| tx.txid().as_bytes().to_vec())
+                .collect();
+
+            MerkleTree::<Sha512Trunc256Sum>::new(&txid_vecs).root()
+        };
+        block.header.tx_merkle_root = tx_merkle_root;
```

Comment:
I'm sure this is a copy/paste from elsewhere, but `NakamotoBlockHeader::empty()` should suffice for this test I believe

Reply:
Done!